From 424a5a02075b01c36e4068f7afbcf061fbfc50cd Mon Sep 17 00:00:00 2001
From: Jan Moeller <Jan.Moeller@governikus.de>
Date: Fri, 22 Aug 2025 10:29:43 +0200
Subject: a11y Windows: Announce change to Accessible.name for focused element

The changeset bcd0a3220348778c0d72faac202efb97f4d6dd07 restricted
notifications of updates to Accessible.name to the role
QAccessible::ComboBox to avoid slowdowns to the application.
This change extends the notification for the currently focused element
so the user is correctly informed.

Task-number: QTBUG-139275
Pick-to: 6.8 6.9 6.10
Change-Id: Ic0daa0b10407780434594b9b5888c9d740c4b80b
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 4161ae0a75acbec8a961b8ef5611acd6d7fc9b16)
---
 .../windows/uiautomation/qwindowsuiamainprovider.cpp        | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git x/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp y/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
index 30471ee68c1eb7868edd21154fd8e35126e8f199..3417a3892115ddf303e612ab5bca62aa1d32dada 100644
--- x/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
+++ y/qtbase/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
@@ -160,9 +160,9 @@ void QWindowsUiaMainProvider::notifyValueChange(QAccessibleValueChangeEvent *eve
 void QWindowsUiaMainProvider::notifyNameChange(QAccessibleEvent *event)
 {
     if (QAccessibleInterface *accessible = event->accessibleInterface()) {
-        // Restrict notification to combo boxes, which need it for accessibility,
-        // in order to avoid slowdowns with unnecessary notifications.
-        if (accessible->role() == QAccessible::ComboBox) {
+        // Restrict notification to combo boxes and the currently focused element, which
+        // need it for accessibility, in order to avoid slowdowns with unnecessary notifications.
+        if (accessible->role() == QAccessible::ComboBox || accessible->state().focused) {
             if (auto provider = providerForAccessible(accessible)) {
                 QComVariant oldVal;
                 QComVariant newVal{ accessible->text(QAccessible::Name) };
